modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/MeResourceManagerTest.java

```diff
@@ -0,0 +1,1514 @@
+        String id = userOld.getId();
+
+        return new Object[][]{
+                {id, NEW_USER_SCIM_OBJECT_STRING_FOR_PATCH_UPDATE, "userName", null,
+                        ResponseCodeConstants.CODE_NOT_IMPLEMENTED}
+        };
+    }
+
+    //NotImplementedException
+    @Test(dataProvider = "dataForUpdateWithPATCHNotImplementedException")
+    public void testUpdateWithPATCHNotImplementedException(String existingId, String scimObjectString,
+                                                           String attributes, String excludeAttributes,
+                                                           int expectedScimResponseStatus)
+            throws BadRequestException, NotFoundException, CharonException {
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
+
+        mockStatic(AbstractResourceManager.class);
+
+        when(AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(SCIM2_ME_ENDPOINT);
+        when(AbstractResourceManager.getEncoder()).thenReturn(new JSONEncoder());
+        when(AbstractResourceManager.getDecoder()).thenReturn(new JSONDecoder());
+        when(AbstractResourceManager.encodeSCIMException(any(NotImplementedException.class)))
+                .thenReturn(getEncodeSCIMExceptionObject(new NotImplementedException()));
+
+        when(userManager.getMe(existingId,
+                ResourceManagerUtil.getAllAttributeURIs(schema))).thenThrow(NotImplementedException.class);
+        SCIMResponse scimResponse = meResourceManager.updateWithPATCH(existingId, scimObjectString,
+                userManager, attributes, excludeAttributes);
+        Assert.assertEquals(scimResponse.getResponseStatus(), expectedScimResponseStatus);
+
+    }
+
+    @DataProvider(name = "dataForUpdateWithPATCHInternalErrorException")
+    public Object[][] dataToUpdateWithPATCHInternalErrorException()
+            throws BadRequestException, CharonException, InternalErrorException {
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
+        JSONDecoder decoder = new JSONDecoder();
+
+        User userOld = decoder.decodeResource(NEW_USER_SCIM_OBJECT_STRING_FOR_PATCH, schema, new User());
+        String id = userOld.getId();
+
+        return new Object[][]{
+                {id, NEW_USER_SCIM_OBJECT_STRING_FOR_PATCH_UPDATE,
+                        "userName", null, ResponseCodeConstants.CODE_INTERNAL_ERROR}
+        };
+    }
+
+    //InternalErrorException
+    @Test(dataProvider = "dataForUpdateWithPATCHInternalErrorException")
+    public void testUpdateWithPATCHInternalErrorException(String existingId, String scimObjectString,
+                                                          String attributes, String excludeAttributes,
+                                                          int expectedScimResponseStatus
+    )
```

Comment:
```suggestion
                                                          int expectedScimResponseStatus)
```

Reply:
Fixed in commit 46d2fbb195d18c319a4e0f4ac4f487ce19a0cafe